src/Microsoft.Tye.Core/ProjectReader.cs

```diff
@@ -30,94 +39,136 @@ public static async Task ReadProjectDetailsAsync(OutputContext output, FileInfo
-                var exitCode = await Process.ExecuteAsync(
-                    $"dotnet",
-                    msbuildCommand,
-                    workingDir: projectFile.DirectoryName,
-                    stdOut: capture.StdOut,
-                    stdErr: capture.StdErr);
-
-                output.WriteDebugLine($"Done running 'dotnet msbuild' exit code: {exitCode}");
-                if (exitCode != 0)
-                {
-                    throw new CommandException("'dotnet msbuild' failed.");
-                }
+                AssemblyLoadContext.Default.Resolving += ResolveAssembly;
+                var result = projectInstance.Build(
+                    targets: new[] { "Restore", "ResolveReferences", "ResolvePackageDependenciesDesignTime", "PrepareResources", "GetAssemblyAttributes", },
+                    loggers: new[] { logger, });
+
+                // If the build fails, we're not really blocked from doing our work.
+                // For now we just log the output to debug. There are errors that occur during
+                // running these targets we don't really care as long as we get the data.
+            }
+            finally
+            {
+                AssemblyLoadContext.Default.Resolving -= ResolveAssembly;
+            }
 
-                var lines = await File.ReadAllLinesAsync(outputFilePath);
-                for (var i = 0; i < lines.Length; i++)
+            // Reading both InformationalVersion and Version is more resilant in the face of build failures.
+            var version = projectInstance.GetProperty("InformationalVersion")?.EvaluatedValue ?? projectInstance.GetProperty("Version").EvaluatedValue;
+            project.Version = version;
+            output.WriteDebugLine($"Found application version: {version}");
+
+            var targetFramework = projectInstance.GetPropertyValue("TargetFramework");
+            project.TargetFramework = targetFramework;
+            output.WriteDebugLine($"Found target framework: {targetFramework}");
+
+            var sharedFrameworks = projectInstance.GetItems("FrameworkReference").Select(i => i.EvaluatedInclude).ToList();
+            project.Frameworks.AddRange(sharedFrameworks.Select(s => new Framework(s)));
+            output.WriteDebugLine($"Found shared frameworks: {string.Join(", ", sharedFrameworks)}");
+
+            output.WriteDebugLine($"Evaluation Took: {sw.Elapsed.TotalMilliseconds}ms");
+
+            // The Microsoft.Build.Locator doesn't handle the loading of other assemblies
+            // that are shipped with MSBuild (ex NuGet).
+            //
+            // This means that the set of assemblies that need special handling depends on the targets
+            // that we run :(
+            //
+            // This is workaround for this limitation based on the targets we need to run
+            // to resolve references and versions.
+            //
+            // See: https://github.com/microsoft/MSBuildLocator/issues/86
```

Comment:
:heart: (Yes, we should do this for you)

Reply:
Cool, let us know if/when that happens.